rust/oak_tests/src/lib.rs

```diff
@@ -1,66 +1,99 @@
 use std::cell::RefCell;
+use std::collections::VecDeque;
 
 #[cfg(test)]
 mod tests;
 
-// Create a test-only implementation of channel_read and channel_write which
-// holds a single message (and which ignores channel handle).  A non-None status
-// value (set with set_status()) will be returned on any channel operation.
-thread_local! {
-    static MESSAGE: RefCell<Vec<u8>> = RefCell::new(Vec::new());
-    static STATUS: RefCell<Option<i32>> = RefCell::new(None);
+struct MockChannel {
+    /// If read_status is set, this status value will be returned for any read
+    /// operations on the mock channel (and |messages| will be left
+    /// undisturbed).
+    pub read_status: Option<i32>,
```

Comment:
Alternatively could make this of type `i32` and set it to `OK` by default when creating the channel?

Reply:
I kinda like the `Option`, it feels like it makes the behaviour clearer (`None =>` act like a real channel, `Some(s) =>` inject a fake status result `s`).